WooCommerce/Classes/ViewRelated/OrdersBadgeController.swift

```diff
@@ -32,31 +32,52 @@ final class OrdersBadgeController {
 private extension OrdersBadgeController {
 
     enum Constants {
-        static let diameter    = CGFloat(10)
         static let borderWidth = CGFloat(1)
-        static let xOffset     = CGFloat(10)
-        static let yOffset     = CGFloat(-6)
+        static let landscapeX  = CGFloat(0)
+        static let xOffset     = CGFloat(14)
+        static let yOffset     = CGFloat(1)
         static let tagOffset   = 999
-        static let width       = CGFloat(17)
-        static let height      = CGFloat(17)
+        static let width       = CGFloat(15)
+        static let extendWidth = CGFloat(23)
+        static let height      = CGFloat(15)
     }
 
     func badge(for tab: WooTab, with text: String) -> BadgeLabel {
-        let returnValue = BadgeLabel(frame: CGRect(x: Constants.xOffset,
+        var width = Constants.width
+        var horizontalPadding = CGFloat(2)
+        var xOffset = Constants.xOffset
+
+        let ninePlus = NSLocalizedString("9+", comment: "A badge with the number of orders. More than nine new orders.")
+        if text == ninePlus {
+            horizontalPadding = CGFloat(4)
+            width = Constants.extendWidth
+        }
+
+        if UIApplication.shared.statusBarOrientation.isLandscape {
```

Comment:
I think the check on `isLandscape` will miss the iPad portrait case, where the tab icon and title also have a horizontal layout (screenshot below). I couldn't find the official documentation on the conditions for horizontal/vertical tab bar item layout, but it seems to have horizontal layout when the screen is landscape OR has [regular horizontal size class](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/adaptivity-and-layout/)

![Simulator Screen Shot - iPad Air 2 - 2019-08-16 at 12 52 51](https://user-images.githubusercontent.com/1945542/63144334-ff887780-c024-11e9-973d-8e2ab5bb9b86.png)

Reply:
Good catch! I forgot to address this with my last set of changes, thanks for reminding me. This is why I prefer .xib files when working with UI. Size classes make it more obvious when I miss cases like this.